presto-main/src/main/java/io/prestosql/Session.java

```diff
@@ -315,7 +340,7 @@ public Session beginTransactionId(TransactionId transactionId, TransactionManage
 
             for (Entry<String, String> property : catalogProperties.entrySet()) {
                 // verify permissions
-                accessControl.checkCanSetCatalogSessionProperty(transactionId, identity, catalogName, property.getKey());
+                accessControl.checkCanSetCatalogSessionProperty(sessionWithTransaction, catalogName, property.getKey());
```

Comment:
revert. instead pass `identity` to `checkCanSetCatalogSessionProperty`

Reply:
`checkCanSetCatalogSessionProperty` requires admin level privileges (write access on entire catalog), it would be a pity to not store that access in audit storage.

Reply:
Passing a session that is not fully formed yet looks error-prone to me. Let's discuss.

Reply:
Agree. See https://github.com/prestosql/presto/pull/171#issuecomment-461359314